### ISSUE TITLE
ctrl+shift+x shortcut is not focusing the right pane.

### ISSUE THREAD

**Issue by manmartgarc** [spyder-ide/spyder#10952]
## Problem Description

The shortcut _switch to explorer_ (Ctrl+Shift+X) does not move the keyboard cursor to the correct window pane. This was working in Spyder 3.7

### What steps reproduce the problem?

1. Install spyder=4.0.0
2. Try using Ctrl+Shift+X
3. Cursor is still in last current pane

### What is the expected output? What do you see instead?
The expected output is that after I hit the shortcut, I can use the keyboard to navigate the Explorer pane, however the cursor remains on the last-used pane.

## Versions

* Spyder version: 4.0.0
* Python version: Python 3.6.9 64-bit
* Qt version: Qt 5.9.6
* PyQt version: PyQt5 5.9.2
* Operating System name/version: Windows 2008ServerR2

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.5.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  None (NOK)
cython >=0.21                :  None (NOK)
IPython >=4.0                :  7.10.1 (OK)
matplotlib >=2.0.0           :  3.1.1 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
numpy >=1.7                  :  1.17.4 (OK)
scipy >=0.17.0               :  1.3.1 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
```

**Comment by jnsebgosselin** [comment]
I can replicate this. The problem is also occurring for the outline explorer.

@goanpeca @ccordoba12 Is keeping the focus on the Editor working as intended or is this a bug?

**Comment by goanpeca** [comment]
I will take a look. What do you think?

**Comment by jnsebgosselin** [comment]
> 
> 
> I will take a look. What do you think?

I think the focus should be given to the main widget of the plugin we are switching to.

**Comment by gerazov** [comment]
I'll add to this and say that when in Console I can't `switch to Editor` using the `Ctrl+Shift+E` shortcut. Also using it to switch to Files, Help, Variable explorer activates these panes (I have them tabbed) but focus remains in the Console.

**Comment by ccordoba12** [comment]
@gerazov, I can't reproduce that on Linux. Is your console undocked?

**Comment by gerazov** [comment]
You're right - it only happens when I have a New Editor window. It works fine in the normal layout even with split views in the Editor.

And for the Files, Outline, Variable explorer I guess it's because they don't have keyboard navigation at all no?

**Comment by ccordoba12** [comment]
@spyder-ide/core-developers, what do you think about giving focus to the panes that users can interact with using the keyboard (like the ones mentioned above by @gerazov)?

**Comment by goanpeca** [comment]
>what do you think about giving focus to the panes that users can interact with using the keyboard (like the ones mentioned above by @gerazov)?

I agree that Spyder should provide a way of using it without a mouse, so anything that allows for this I am 👍.

Also panels should allow for `<Tab>`navigation to move between the different widgets

**Comment by ccordoba12** [comment]
I agree. Leaving this for 4.2.0 because it could require important changes to several panes (most probably not, but I want to be on the safe side here).

**Comment by jitseniesen** [comment]
I seem to remember that fairly recently there was a discussion that for some panes like the new plot pane, we want Ctrl-Shift-G only to raise the pane if it is hidden underneath some other pane. You'd have to press it twice to switch focus. However, I can't quickly find that discussion anymore. Did I just dream it?

**Comment by ccordoba12** [comment]
> Did I just dream it?

No, you didn't. I also remember it too, but I don't recall its number.

**Comment by jnsebgosselin** [comment]
> 
> 
> > Did I just dream it?
> 
> No, you didn't. I also remember it too, but I don't recall its number.

It was on Gitter. This behavior is already implementer for the plot pane in PR #11684

![image](https://user-images.githubusercontent.com/10170372/81500230-cb35a500-929e-11ea-80e7-a3e8abe29015.png)

**Comment by pyested** [comment]
Any updates on this? I'm using v6.0.0 and still the shortcut _switch to explorer_ (Ctrl+Shift+X) doesn't focus the explorer pane

**Comment by dalthviz** [comment]
Hi there @pyested as an update there has been some discussions over the focus behavior for a couple of other panes like Plots (https://github.com/spyder-ide/spyder/issues/22381) and Help (https://github.com/spyder-ide/spyder/issues/22189). Over https://github.com/spyder-ide/spyder/issues/22189#issuecomment-2248269160 a possible idea to enable users to select if they want or not the panes to get focus was discussed (which I think is partailly related with the one that was left here 😅). Will move this to the appropriate milestone following the other issues :+1:
